exercises/diffie-hellman/tests/diffie-hellman.rs

```diff
@@ -0,0 +1,58 @@
+extern crate diffie_hellman;
+
+use diffie_hellman::*;
+
+#[test]
+fn test_private_key_in_range_key() {
+    let primes: Vec<u64> = vec![5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 773, 9091,
+                                38321, 6131329];
+    let private_keys: Vec<u64> = primes.iter().map(|x| private_key(*x)).collect();
+
+    for i in 0..primes.len() {
+        assert!(1 < private_keys[i] && private_keys[i] < primes[i]);
```

Comment:
On https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange I see a quote that says:

> Alice picks a random natural number a, where 1 ≤ a < n, and sends g<sup>a</sup> to Bob.

So, it seems like 1 would be valid, but this line disallows it. Is 1 in fact allowed, or did I misunderstand?

Reply:
Hmm I think then the [description](https://github.com/hekrause/problem-specifications/blob/master/exercises/diffie-hellman/description.md) is wrong.

Reply:
@petertseng 
Ah I think in this section of the [wiki](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange) they are taking ```a``` from a pool of known numbers.
We are working here with primes where 1 is no prime.